spec/operators/bufferToggle-spec.ts

```diff
@@ -1,352 +1,414 @@
 import { bufferToggle, mergeMap, mapTo } from 'rxjs/operators';
-import { hot, cold, expectObservable, expectSubscriptions } from '../helpers/marble-testing';
+import { TestScheduler } from 'rxjs/testing';
+import { observableMatcher } from '../helpers/observableMatcher';
 
 declare function asDiagram(arg: string): Function;
 
 /** @test {bufferToggle} */
 describe('bufferToggle operator', () => {
+  let testScheduler: TestScheduler;
+
+  beforeEach(() => {
+    testScheduler = new TestScheduler(observableMatcher);
+  });
+
   asDiagram('bufferToggle')('should emit buffers using hot openings and hot closings', () => {
-    const e1 =   hot('---a---b---c---d---e---f---g---|');
-    const e2 =   hot('--o------------------o---------|');
-    const e3 =   hot('---------c---------------c-----|');
-    const expected = '---------x---------------y-----|';
-    const values = {
-      x: ['a', 'b'],
-      y: ['f'],
-    };
-
-    const result = e1.pipe(bufferToggle(e2, (x: any) => e3));
-
-    expectObservable(result).toBe(expected, values);
+    testScheduler.run(({ hot, expectObservable }) => {
+      const e1 = hot('  ---a---b---c---d---e---f---g---|');
+      const e2 = hot('  --o------------------o---------|');
+      const e3 = hot('  ---------c---------------c-----|');
+      const expected = '---------x---------------y-----|';
+      const values = {
+        x: ['a', 'b'],
+        y: ['f'],
+      };
+
+      const result = e1.pipe(bufferToggle(e2, (x: any) => e3));
+
+      expectObservable(result).toBe(expected, values);
+    });
   });
 
   it('should emit buffers that are opened by an observable from the first argument ' +
       'and closed by an observable returned by the function in the second argument',
   () => {
-    const e1 =   hot('-----a----b----c----d----e----f----g----h----i----|');
-    const e2 =  cold('-------------x-------------y--------------z-------|');
-    const e3 =               cold('---------------(j|)');
-    //                                         ---------------(j|)
-    //                                                        ---------------(j|)
-    const expected = '----------------------------q-------------r-------(s|)';
-
-    const values = {
-      q: ['c', 'd', 'e'],
-      r: ['f', 'g', 'h'],
-      s: ['i']
-    };
-    const innerVals = ['x', 'y', 'z'];
-
-    expectObservable(e1.pipe(
-      bufferToggle(e2, (x: string) => {
-        expect(x).to.equal(innerVals.shift());
-        return e3;
-      }))
-    ).toBe(expected, values);
+    testScheduler.run(({ hot, cold, expectObservable }) => {
+      const e1 = hot('  -----a----b----c----d----e----f----g----h----i----|');
+      const e2 = cold(' -------------x-------------y--------------z-------|');
+      const e3 = cold('              ---------------(j|)');
+      //                                         ---------------(j|)
+      //                                                        ---------------(j|)
```

Comment:
Alignment.

Reply:
not sure what to fix on this one

Reply:
Throughout the tests, there are places in which comment marbles are used to help show how things line up. AFAICT, in this test the comment marbles are misaligned. It seems to me that they should be aligned with the values emitted by `e2`, like this:

```
it('should emit buffers that are opened by an observable from the first argument ' +
    'and closed by an observable returned by the function in the second argument',
() => {
  testScheduler.run(({ hot, cold, expectObservable }) => {
    const e1 = hot('  -----a----b----c----d----e----f----g----h----i----|');
    const e2 = cold(' -------------x-------------y--------------z-------|');
    const e3 = cold('              ---------------(j|)');
    //                                           ---------------(j|)
                                                 ^
                                                 here
    //                                                          ---------------(j|)
                                                                ^
                                                                here
    const expected = '----------------------------q-------------r-------(s|)';

    const values = {
      q: ['c', 'd', 'e'],
      r: ['f', 'g', 'h'],
      s: ['i']
    };
    const innerVals = ['x', 'y', 'z'];

    expectObservable(e1.pipe(
      bufferToggle(e2, (x: string) => {
        expect(x).to.equal(innerVals.shift());
        return e3;
      }))
    ).toBe(expected, values);
  });
});
```

Reply:
I think if there are going to be comment marbles - that serve as guidelines so that the content and alignment of actual marble diagrams makes more sense, we should try to make sure they are aligned, too. And, hopefully, in a way that will survive Prettier's (eventual) reformatting.

Reply:
ok, thanks - I pushed that fix